v4/source/plugins.yaml

```diff
@@ -440,11 +443,13 @@
               type: object
               required:
                 - id
-                - version
```

Comment:
Wondering why are we removing this?

Reply:
Hi, because `version` is defined as "*required*", but there isn't even a "*version property*" defined in the following properties array. Only Id is defined. So, it can't be required if it isn't even a valid property. You know what I mean?
For the definition to be valid, either `version` should be removed from the required string, or added as a property to the following properties array.

I just wen't through the code again. While "version" would indeed be a valid field according to the struct definition [here](https://github.com/mattermost/mattermost-server/blob/9258a08b8da0fcc69b3b2e3a496d532023af11da/model/marketplace_plugin.go#L114), it is always [set to the empty string](https://github.com/mattermost/mattermost-server/blob/9258a08b8da0fcc69b3b2e3a496d532023af11da/api4/plugin.go#L156) before installing the plugin to always get the latest version (according to the comment there).

So, the version property could be passed to the endpoint, but it won't be used currently, as it seems. 
I have to admit, I don't know if parsing fails if it is not passed. But adding version to the property array to fix the API definition instead of removing it from required might still be the better option, even if it isn't used.